[TRTLLM-10590][feat] Eagle3 Specdec UX improvements

venkywonka opened the pull request:
## Description

<!--
Please explain the issue and the solution in short.
-->
 
### 1. Make `draft_vocab_size` optional for Eagle3, and default to `target_vocab_size`

- Currently to launch eagle3 in trtllm-serve, user needs to explicitly supply `draft_vocab_size` as part of the nested `speculative_decoding_config` in the config.yaml.
- vLLM assumes the `draft_vocab_size == target_vocab_size` in the default, unspecified situation without throwing an error. Following the same behavior would improve usability and keeping our API tight with vLLM for interop. 

### 2. Deprecate `decoding_type: Eagle` for pytorch backend, introduce `decoding_type: Eagle3`.

- `Eagle` (v1, v2) are fundamentally different from `Eagle3` - and should not be conflated.
- Related discussion: https://nvidia.slack.com/archives/C059LSY62BT/p1766398949447109 
- users can potentially get confused [here](https://nvidia.github.io/TensorRT-LLM/features/speculative-decoding.html#usage-with-trtllm-bench-and-trtllm-serve) - where Eagle seems to imply Eagle3
- It is nice that a small clarification is mentioned "Eagle (for Eagle3)" - but it would be better if it was clearly named decoding_type: Eagle3 to start with:

<img width="719" height="482" alt="image" src="https://github.com/user-attachments/assets/8fc6d975-7146-4d38-a1a6-d883225381af" />


- Currently, TRTLLM pytorch backend *doe not support Eagle* and only supports Eagle 3. The Legacy (TRT) backend is exactly opposite - supports Eagle and does not support Eagle 3.
- This PR makes this clear by:
  - For the default, pytorch backend: Printing a warning for users providing `decoding_type: Eagle` to switch over - so its non-breaking - and explicitly inform them that TRTLLM pytorch backend treats `decoding_type: Eagle` as `Eagle3` and therefore expects Eagle 3 draft checkpoint. The user must be explicitly noted that providing an Eagle draft checkpoint will result in errors - currently this is rather implicit and not documented clearly.
  - For TRT legacy backend: Specifying `decoding_type: Eagle3` will raise an error. `decoding_type: Eagle` will continue to work as usual.


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Eagle3 decoding support now available with PyTorch backend
  * Added Eagle3 configuration class for speculative decoding

* **Documentation**
  * Updated documentation to reflect Eagle3 as the preferred decoding option

* **Bug Fixes**
  * Improved draft vocabulary size handling with automatic defaults and warnings

* **Tests**
  * Added Eagle3 configuration and backend validation tests

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

<!--
Please write the PR title by following this template:

**[JIRA ticket/NVBugs ID/GitHub issue/None][type] Summary**

Valid ticket formats:
  - JIRA ticket: [TRTLLM-1234] or [FOOBAR-123] for other FOOBAR project
  - NVBugs ID: [https://nvbugs/1234567]
  - GitHub issue: [#1234]
  - No ticket: [None]

Valid types (lowercase): [fix], [feat], [doc], [infra], [chore], etc.

Examples:
  - [TRTLLM-1234][feat] Add new feature
  - [https://nvbugs/1234567][fix] Fix some bugs
  - [#1234][doc] Update documentation
  - [None][chore] Minor clean-up

Alternative (faster) way using CodeRabbit AI:

**[JIRA ticket/NVBugs ID/GitHub issue/None] @coderabbitai title**

NOTE: "@coderabbitai title" will be replaced by the title generated by CodeRabbit AI, that includes the "[type]" and title.
For more info, see /.coderabbit.yaml.

-->


## Test Coverage

<!--
Please list clearly what are the relevant test(s) that can safeguard the changes in the PR. This helps us to ensure we have sufficient test coverage for the PR.
-->

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.


- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

`/bot [-h] ['run', 'kill', 'skip', 'reuse-pipeline'] ...`

Provide a user friendly way for developers to interact with a Jenkins server.

Run `/bot [-h|--help]` to print this help message.

See details below for each supported subcommand.

<details>

`run  [--reuse-test (optional)pipeline-id --disable-fail-fast --skip-test --stage-list "A10-PyTorch-1, xxx" --gpu-type "A30, H100_PCIe" --test-backend "pytorch, cpp" --add-multi-gpu-test --only-multi-gpu-test --disable-multi-gpu-test --post-merge --extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx" --detailed-log --debug(experimental)]`

Launch build/test pipelines. All previously running jobs will be killed.

`--reuse-test (optional)pipeline-id ` *(OPTIONAL)* : Allow the new pipeline to reuse build artifacts and skip successful test stages from a specified pipeline or the last pipeline if no pipeline-id is indicated. If the Git commit ID has changed, this option will be always ignored. The DEFAULT behavior of the bot is to reuse build artifacts and successful test results from the last pipeline.

`--disable-reuse-test ` *(OPTIONAL)* : Explicitly prevent the pipeline from reusing build artifacts and skipping successful test stages from a previous pipeline. Ensure that all builds and tests are run regardless of previous successes.

`--disable-fail-fast ` *(OPTIONAL)* : Disable fail fast on build/tests/infra failures.

`--skip-test ` *(OPTIONAL)* : Skip all test stages, but still run build stages, package stages and sanity check stages. Note: Does **NOT** update GitHub check status.

`--stage-list "A10-PyTorch-1, xxx"` *(OPTIONAL)* : Only run the specified test stages. Examples: "A10-PyTorch-1, xxx". Note: Does **NOT** update GitHub check status.

`--gpu-type "A30, H100_PCIe"` *(OPTIONAL)* : Only run the test stages on the specified GPU types. Examples: "A30, H100_PCIe". Note: Does **NOT** update GitHub check status.

`--test-backend "pytorch, cpp"` *(OPTIONAL)* : Skip test stages which don't match the specified backends. Only support [pytorch, cpp, tensorrt, triton]. Examples: "pytorch, cpp" (does not run test stages with tensorrt or triton backend). Note: Does **NOT** update GitHub pipeline status.

`--only-multi-gpu-test ` *(OPTIONAL)* : Only run the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--disable-multi-gpu-test ` *(OPTIONAL)* : Disable the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--add-multi-gpu-test ` *(OPTIONAL)* : Force run the multi-GPU tests in addition to running L0 pre-merge pipeline.

`--post-merge ` *(OPTIONAL)* : Run the L0 post-merge pipeline instead of the ordinary L0 pre-merge pipeline.

`--extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx"` *(OPTIONAL)* : Run the ordinary L0 pre-merge pipeline and specified test stages. Examples: --extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx".

`--detailed-log ` *(OPTIONAL)* : Enable flushing out all logs to the Jenkins console. This will significantly increase the log volume and may slow down the job.

`--debug ` *(OPTIONAL)* : **Experimental feature**. Enable access to the CI container for debugging purpose. Note: Specify exactly one stage in the `stage-list` parameter to access the appropriate container environment. Note: Does **NOT** update GitHub check status.

For guidance on mapping tests to stage names, see `docs/source/reference/ci-overview.md`
and the `scripts/test_to_stage_mapping.py` helper.

### kill

`kill  `

Kill all running builds associated with pull request.

### skip

`skip --comment COMMENT `

Skip testing for latest commit on pull request. `--comment "Reason for skipping build/test"` is required. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

### reuse-pipeline

`reuse-pipeline `

Reuse a previous pipeline to validate current commit. This action will also kill all currently running builds associated with the pull request. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

</details>

venkywonka committed:
Fix EAGLE3 draft_vocab_size fallback

Treat Eagle3-suffixed architectures as EAGLE3 even when draft_vocab_size is missing, and default missing draft_vocab_size to vocab_size with a warning.

Signed-off-by: Venky Ganesh <23023424+venkywonka@users.noreply.github.com>

venkywonka committed:
Test: omit draft_vocab_size to cover EAGLE3 fallback

The EAGLE3 dummy configs in test_eagle3 set draft_vocab_size == vocab_size; omit the field so the new defaulting behavior is exercised.

Signed-off-by: Venky Ganesh <23023424+venkywonka@users.noreply.github.com>

venkywonka committed:
Docs: shorten EAGLE3 draft_vocab_size note

Document the draft_vocab_size defaulting behavior concisely.

Signed-off-by: Venky Ganesh <23023424+venkywonka@users.noreply.github.com>

venkywonka commented:
/bot run --disable-fail-fast

tensorrt-cicd commented:
[PR_Github #28985](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/28985/) [ run ] triggered by Bot. Commit: `10a4571`

arysef commented on tests/unittest/_torch/speculative/test_eagle3.py:
```diff
@@ -433,7 +432,6 @@ def test_multi_eagle3(use_one_model: bool):
         'transformers_version': '4.52.4',
         'use_cache': True,
         'vocab_size': 128256,
-        'draft_vocab_size': 128256
```
Nit: instead of removing draft_vocab_size in the existing tests, would it make sense to add a new test case that doesn't have it?

venkywonka committed:
Add Eagle3 decoding_type

Introduce speculative_config.decoding_type: Eagle3 for the PyTorch backend, warn when using Eagle as an alias, and reject Eagle3 on the TensorRT backend. Update docs/examples and add unit tests.

Signed-off-by: Venky Ganesh <23023424+venkywonka@users.noreply.github.com>

venkywonka committed:
Tighten Eagle/Eagle3 messages

De-duplicate draft_vocab_size fallback warning text and clarify that decoding_type: Eagle is a PyTorch-backend alias for Eagle3 (EAGLE v1/v2 draft checkpoints are incompatible).

Signed-off-by: Venky Ganesh <23023424+venkywonka@users.noreply.github.com>

venkywonka committed:
Test: add dedicated unit test for draft_vocab_size fallback

Add test_eagle3_defaults_draft_vocab_size_when_missing to explicitly
test the fallback behavior when draft_vocab_size is missing from
pretrained config.

Restore draft_vocab_size in existing test configs (test_deepseek_eagle3,
test_multi_eagle3) rather than relying on the fallback path.

Update test imports to use Eagle3DecodingConfig.

Signed-off-by: Venky Ganesh <23023424+venkywonka@users.noreply.github.com>

venkywonka committed:
Export Eagle3DecodingConfig and update docs/examples/tests

- Export Eagle3DecodingConfig from llmapi
- Add _decoding_type_alias tracking for Eagle→Eagle3 mapping on PyTorch
- Update from_dict to map 'Eagle' to Eagle3DecodingConfig on PyTorch backend
- Show deprecation warning when 'Eagle' is used on PyTorch backend
- Reject 'Eagle3' on TensorRT backend with clear error message
- Update docs and examples to use Eagle3DecodingConfig
- Update test imports to Eagle3DecodingConfig

Signed-off-by: Venky Ganesh <23023424+venkywonka@users.noreply.github.com>

venkywonka committed:
apply review changes

Signed-off-by: Venky Ganesh <23023424+venkywonka@users.noreply.github.com>

tensorrt-cicd commented:
[PR_Github #29603](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/29603/) [ run ] triggered by Bot. Commit: `2218e4d`

QiJune reviewed (approved):
LGTM

anish-shanbhag commented on tensorrt_llm/_torch/models/modeling_auto.py:
```diff
@@ -29,7 +30,8 @@ def from_config(
                                         "")  # Strip the appended EAGLE3
-        if hasattr(config.pretrained_config, "draft_vocab_size"):
+        if hasattr(config.pretrained_config,
+                   "draft_vocab_size") or has_eagle3_suffix:
```
Just curious, why is this change necessary?

anish-shanbhag commented on tensorrt_llm/_torch/models/modeling_speculative.py:
```diff
@@ -26,6 +27,17 @@
+    if hasattr(config, "draft_vocab_size"):
+        return
+
+    warnings.warn(
```
Should we use `logger.warning` instead?

venkywonka committed:
Improve Eagle3 detection comments and use logger

Signed-off-by: Venky Ganesh <23023424+venkywonka@users.noreply.github.com>

venkywonka commented:
/bot run

tensorrt-cicd commented:
[PR_Github #30547](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/30547/) [ run ] triggered by Bot. Commit: `21b499a`

venkywonka commented:
/bot run --disable-fail-fast

venkywonka committed:
Fix undefined draft_vocab_size in Eagle3ForCausalLM

Signed-off-by: Venky Ganesh <23023424+venkywonka@users.noreply.github.com>

tensorrt-cicd commented:
[PR_Github #30632](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/30632/) [ run ] triggered by Bot. Commit: `f3a577d`

tensorrt-cicd commented:
[PR_Github #30632](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/30632/) [ run ] completed with state `SUCCESS`. Commit: `f3a577d` 
[/LLM/main/L0_MergeRequest_PR pipeline #23639](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/23639/) completed with status: 'FAILURE'

⚠️ **Action Required:**
- Please check the failed tests and fix your PR
- If you cannot view the failures, ask the CI triggerer to share details
- Once fixed, request an NVIDIA team member to trigger CI again

venkywonka commented:
/bot kill

venkywonka commented:
/bot run --stage-list "H100_PCIe-PyTorch-1"

tensorrt-cicd commented:
[PR_Github #30674](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/30674/) [ run ] triggered by Bot. Commit: `e9e4ff5`

tensorrt-cicd commented:
[PR_Github #30682](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/30682/) [ run ] triggered by Bot. Commit: `e9e4ff5`

venkywonka commented:
/bot run --stage-list "H100_PCIe-PyTorch-1"

tensorrt-cicd commented:
[PR_Github #30744](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/30744/) [ run ] completed with state `SUCCESS`. Commit: `6fcc572` 
[/LLM/main/L0_MergeRequest_PR pipeline #23726 `(Partly Tested)`](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/23726/) completed with status: 'FAILURE'

⚠️ **Action Required:**
- Please check the failed tests and fix your PR
- If you cannot view the failures, ask the CI triggerer to share details
- Once fixed, request an NVIDIA team member to trigger CI again

venkywonka committed:
Fix Eagle3 detection to check draft_vocab_size attribute

Signed-off-by: Venky Ganesh <23023424+venkywonka@users.noreply.github.com>

tensorrt-cicd commented:
[PR_Github #30817](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/30817/) [ run ] triggered by Bot. Commit: `429b650`

venkywonka committed:
Clarify MistralLarge3Eagle uses EAGLE-3 despite naming

Signed-off-by: Venky Ganesh <23023424+venkywonka@users.noreply.github.com>

venkywonka commented:
/bot run

tensorrt-cicd commented:
[PR_Github #30876](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/30876/) [ run ] triggered by Bot. Commit: `364ea9a`

venkywonka committed:
Fix _ensure_draft_vocab_size to handle None values

Signed-off-by: Venky Ganesh <23023424+venkywonka@users.noreply.github.com>

venkywonka commented:
/bot run

tensorrt-cicd commented:
[PR_Github #30917](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/30917/) [ run ] triggered by Bot. Commit: `7f9e013`

tensorrt-cicd commented:
[PR_Github #30966](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/30966/) [ run ] triggered by Bot. Commit: `7f9e013`

venkywonka commented:
/bot run

venkywonka committed:
Migrate tests from EagleDecodingConfig to Eagle3DecodingConfig

Replace all usages of EagleDecodingConfig with Eagle3DecodingConfig
across test files for consistency. Eagle3DecodingConfig is the
preferred class with decoding_type="Eagle3" instead of the legacy
"Eagle".

Also fix warning test to use caplog instead of pytest.warns since
the code uses logger.warning() not warnings.warn().

Files updated:
- tests/unittest/_torch/speculative/test_eagle3.py
- tests/integration/defs/accuracy/test_llm_api_pytorch.py
- tests/integration/defs/accuracy/test_llm_api.py
- tests/integration/defs/accuracy/test_cli_flow.py
- tests/integration/defs/test_e2e.py
- tests/integration/defs/examples/test_ad_speculative_decoding.py
- tests/unittest/llmapi/test_llm.py

Signed-off-by: Venky Ganesh <23023424+venkywonka@users.noreply.github.com>

venkywonka committed:
Remove unused _ensure_draft_vocab_size test and import

Signed-off-by: Venky Ganesh <23023424+venkywonka@users.noreply.github.com>

tensorrt-cicd commented:
[PR_Github #31276](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/31276/) [ run ] triggered by Bot. Commit: `438fe8a`

venkywonka committed:
revert back Eagle3DecodingConfig migrations for trt backend tests

Signed-off-by: Venky Ganesh <23023424+venkywonka@users.noreply.github.com>

tensorrt-cicd commented:
[PR_Github #31626](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/31626/) [ kill ] triggered by Bot. Commit: `dda588e`

venkywonka committed:
modify data reference sources

Signed-off-by: Venky Ganesh <23023424+venkywonka@users.noreply.github.com>

tensorrt-cicd commented:
[PR_Github #31902](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/31902/) [ run ] triggered by Bot. Commit: `c558e75`

tensorrt-cicd commented:
[PR_Github #31902](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/31902/) [ run ] completed with state `SUCCESS`. Commit: `c558e75` 
[/LLM/main/L0_MergeRequest_PR pipeline #24705](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/24705/) completed with status: 'FAILURE'

⚠️ **Action Required:**
- Please check the failed tests and fix your PR
- If you cannot view the failures, ask the CI triggerer to share details
- Once fixed, request an NVIDIA team member to trigger CI again

venkywonka committed:
add Eagle3 to more reference data

Signed-off-by: Venky Ganesh <23023424+venkywonka@users.noreply.github.com>

tensorrt-cicd commented:
[PR_Github #32019](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/32019/) [ run ] triggered by Bot. Commit: `97a443d`

tensorrt-cicd commented:
[PR_Github #32019](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/32019/) [ run ] completed with state `SUCCESS`. Commit: `97a443d` 
[/LLM/main/L0_MergeRequest_PR pipeline #24812](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/24812/) completed with status: 'FAILURE'

⚠️ **Action Required:**
- Please check the failed tests and fix your PR
- If you cannot view the failures, ask the CI triggerer to share details
- Once fixed, request an NVIDIA team member to trigger CI again

Commit:
modify all eagle reference accuracy scores to eagle3

Signed-off-by: Venky Ganesh <gvenkatarama@eos0354.eos.clusters.nvidia.com>

venkywonka commented:
/bot run --disable-fail-fast

tensorrt-cicd commented:
[PR_Github #32192](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/32192/) [ run ] triggered by Bot. Commit: `5b3e157`

tensorrt-cicd commented:
[PR_Github #32323](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/32323/) [ kill ] triggered by Bot. Commit: `5b3e157`

venkywonka committed:
precommit is happy

Signed-off-by: Venky Ganesh <23023424+venkywonka@users.noreply.github.com>

tensorrt-cicd commented:
[PR_Github #32344](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/32344/) [ run ] triggered by Bot. Commit: `13564cc`

tensorrt-cicd commented:
[PR_Github #32354](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/32354/) [ run ] completed with state `FAILURE`. Commit: `13564cc` 
[/LLM/main/L0_MergeRequest_PR pipeline #25070](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/25070/) completed with status: 'FAILURE'

⚠️ **Action Required:**
- Please check the failed tests and fix your PR
- If you cannot view the failures, ask the CI triggerer to share details
- Once fixed, request an NVIDIA team member to trigger CI again

tensorrt-cicd commented:
[PR_Github #32359](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/32359/) [ run ] completed with state `SUCCESS`. Commit: `13564cc` 
[/LLM/main/L0_MergeRequest_PR pipeline #25074](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/25074/) completed with status: 'FAILURE'

⚠️ **Action Required:**
- Please check the failed tests and fix your PR
- If you cannot view the failures, ask the CI triggerer to share details
- Once fixed, request an NVIDIA team member to trigger CI again

venkywonka commented:
/bot run

tensorrt-cicd commented:
[PR_Github #32371](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/32371/) [ run ] triggered by Bot. Commit: `13564cc`

venkywonka commented:
/bot run

tensorrt-cicd commented:
[PR_Github #32802](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/32802/) [ run ] completed with state `SUCCESS`. Commit: `13564cc` 
[/LLM/main/L0_MergeRequest_PR pipeline #25380](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/25380/) completed with status: 'FAILURE'

⚠️ **Action Required:**
- Please check the failed tests and fix your PR
- If you cannot view the failures, ask the CI triggerer to share details
- Once fixed, request an NVIDIA team member to trigger CI again

tensorrt-cicd commented:
[PR_Github #32816](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/32816/) [ run ] triggered by Bot. Commit: `13564cc`

venkywonka commented:
/bot run

venkywonka commented:
/bot run

tensorrt-cicd commented:
[PR_Github #33078](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/33078/) [ run ] triggered by Bot. Commit: `fb4ff37`

tensorrt-cicd commented:
[PR_Github #33078](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/33078/) [ run ] completed with state `SUCCESS`. Commit: `fb4ff37` 
[/LLM/main/L0_MergeRequest_PR pipeline #25569](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/25569/) completed with status: 'SUCCESS'
